api/views/api.py

```diff
@@ -11,18 +13,41 @@ def get_data(request):
         if request is None:
             raise Exception("Houston, we have a problem.")
 
-        search = request.GET.get("search")
-        page = request.GET.get("page")
-        per_page = request.GET.get("per_page")
+        page = request.GET.get("page", 1)
+        per_page = request.GET.get("per_page", 12)
 
-        print("options", search, page, per_page)
+        search = request.GET.get("search")
 
-        products = Product.objects.fully_loaded_objects()
-        serializedProducts = [
-            ProductSerializer(product).to_json() for product in products
+        if search is not None:
+            keywords = (
+                SearchProductKeywords.objects.filter(keywords__icontains=search)
+                .select_related("product")
+                .select_related("product__product_price")
+                .prefetch_related(
+                    "product__productimage_set",
+                    "product__productstock_set",
+                    "product__productstock_set__color",
+                )
+            )
+            products = [keyword.product for keyword in keywords]
+
+        else:
+            products = Product.objects.fully_loaded_objects()
+
+        paginator = Paginator(products, per_page)
+        products_for_serialization = paginator.get_page(page)
+
+        num_of_pages = math.ceil(len(products) / int(per_page))
+
+        if int(page) > num_of_pages or int(page) == 0:
```

Comment:
i was going for empty arrays, instead of returning the first or last page... to me, that feels misleading- requesting a page that doesn't exist and getting data for the last or first page that does exist- i was thinking an empty array would be more expected... but if that is highly unusual, i can update

Reply:
empty works :)